src/waypoint.jsx

```diff
@@ -333,13 +352,18 @@ export default class Waypoint extends React.Component {
    * @return {Object}
    */
   render() {
+    if (this.props.children) {
+      return <div ref={this.refElement}>{this.props.children}</div>;
```

Comment:
Bummer, turns out it doesn't work: `this._ref.getBoundingClientRect is not a function`. Looks like the `ref` attached to the HOC doesn't access the underlying DOM node after all 😕 . 

Thoughts on sticking with the original `div` implementation? While unfortunate in that it adds an extra element and could interfere with CSS, it won't break anyone's existing implementation and such issues are often easy to detect and fix. The other possibilities add a lot of mental overhead and complexity.